images.go

```diff
@@ -47,6 +47,11 @@ type Image struct {
 	// or canonical references.
 	Names []string `json:"names,omitempty"`
 
+	// NamesHistory is an optional set of Names the image had in the past. The
+	// contained names are free from any duplicates, whereas the newest entry
+	// is the first one.
+	NamesHistory []string `json:"names-history,omitempty"`
```

Comment:
Currently (and especially with the changes what I expect to happen for `RepoDigests`), `Names` can contain both name:tag and name@digest values. So, `NamesHistory` would currently contain both as well. Is that OK?

(Ideally, I’d like be to insist on strongly typed values like `reference.NamedTagged`, encoding the semantics in the type system, but with `Names []string` that doesn’t seem reasonable right now.)

Reply:
> Currently (and especially with the changes what I expect to happen for `RepoDigests`), `Names` can contain both name:tag and name@digest values. So, `NamesHistory` would currently contain both as well. Is that OK?

I think so.

> (Ideally, I’d like be to insist on strongly typed values like `reference.NamedTagged`, encoding the semantics in the type system, but with `Names []string` that doesn’t seem reasonable right now.)

Yeah, we could change both `Names` and `NamesHistory` to be forcibly valid references, but I wanted to add the change as non-breaking as possible.

Reply:
ACK.